### ISSUE TITLE
Getting defaultSession crashes electron app

### ISSUE THREAD

**Issue by alchen** [electron/electron#5356]
* Electron version: Tried on v0.36.6 and v0.37.8
* Operating system: OS X 10.11.5 Beta 3

I'm trying to implement automatic proxy detections and use it in the main process -- the renderer process appears to use the system settings automatically from Chromium.

I found [this package](https://github.com/felicienfrancois/node-electron-proxy-agent) that seems to do what I want. (It'd be so very nice if there's a cleaner way to go about this...)

Except this line crashes Electron immediately upon launching on my machine: `var session = require('session').defaultSession;`

With the following message:

```
[5436:0429/195645:FATAL:resource_dispatcher_host_impl.cc(1304)] Check failed: ContainsKey(active_resource_contexts_, resource_context). 
0   Electron Framework                  0x0000000101e8a723 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 391203
1   Electron Framework                  0x0000000101ea0be9 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 482537
2   Electron Framework                  0x0000000102626385 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 8369285
3   Electron Framework                  0x00000001026260e2 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 8368610
4   Electron Framework                  0x0000000102625a88 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 8366984
5   Electron Framework                  0x000000010243aa9e _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 6355870
6   Electron Framework                  0x0000000102c7f2ed _CTFontManagerUnregisterFontForData + 3443693
7   Electron Framework                  0x0000000102c73803 _CTFontManagerUnregisterFontForData + 3395843
8   Electron Framework                  0x0000000102c73912 _CTFontManagerUnregisterFontForData + 3396114
9   Electron Framework                  0x0000000102c75ece _CTFontManagerUnregisterFontForData + 3405774
10  Electron Framework                  0x0000000102c762c1 _CTFontManagerUnregisterFontForData + 3406785
11  Electron Framework                  0x0000000102c761cd _CTFontManagerUnregisterFontForData + 3406541
12  Electron Framework                  0x0000000102c75a2c _CTFontManagerUnregisterFontForData + 3404588
13  Electron Framework                  0x0000000102c758dd _CTFontManagerUnregisterFontForData + 3404253
14  Electron Framework                  0x0000000102c711db _CTFontManagerUnregisterFontForData + 3386075
15  Electron Framework                  0x0000000101e7f6c8 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 346056
16  Electron Framework                  0x0000000102b5c335 _CTFontManagerUnregisterFontForData + 2251829
17  Electron Framework                  0x0000000101e7fa79 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 347001
18  Electron Framework                  0x0000000101eba453 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 587091
19  Electron Framework                  0x0000000101ea8f5d _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 516189
20  Electron Framework                  0x00000001024cdb68 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 6958184
21  Electron Framework                  0x00000001024cdc77 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 6958455
22  Electron Framework                  0x0000000101eda4f8 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 718328
23  Electron Framework                  0x0000000101ed658c _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 702092
24  libsystem_pthread.dylib             0x00007fff9045499d _pthread_body + 131
25  libsystem_pthread.dylib             0x00007fff9045491a _pthread_body + 0
26  libsystem_pthread.dylib             0x00007fff90452351 thread_start + 13
```

Though if I used `var ses = session.fromPartition('x');` instead (where `x` is anything, persist or not), this error will be postponed until the Electron application quits, and everything seems to run otherwise fine. The error message is just slight different like below:

```
[5470:0429/195708:FATAL:resource_dispatcher_host_impl.cc(560)] Check failed: ContainsKey(active_resource_contexts_, context). 
0   Electron Framework                  0x0000000100eab723 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 391203
1   Electron Framework                  0x0000000100ec1be9 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 482537
2   Electron Framework                  0x0000000101643b25 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 8354853
3   Electron Framework                  0x000000010170a64c _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 9168716
4   Electron Framework                  0x0000000100e1dc19 _ZN9brightray14BrowserContext15ResourceContextD2Ev + 9
5   Electron Framework                  0x0000000100e1db89 _ZN9brightray14BrowserContext15ResourceContextD1Ev + 9
6   Electron Framework                  0x0000000100e1db9e _ZN9brightray14BrowserContext15ResourceContextD1Ev + 30
7   Electron Framework                  0x0000000100e1e4ef _ZN4base12DeleteHelperIN9brightray14BrowserContext15ResourceContextEE8DoDeleteEPKv + 15
8   Electron Framework                  0x0000000100eabc3a _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 392506
9   Electron Framework                  0x0000000100eca7b9 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 518329
10  Electron Framework                  0x0000000100ecacf8 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 519672
11  Electron Framework                  0x0000000100ea0ac1 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 347073
12  Electron Framework                  0x0000000100edb453 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 587091
13  Electron Framework                  0x0000000100ec9f5d _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 516189
14  Electron Framework                  0x00000001014eeb68 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 6958184
15  Electron Framework                  0x00000001014eec77 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 6958455
16  Electron Framework                  0x0000000100efb4f8 _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 718328
17  Electron Framework                  0x0000000100ef758c _ZN4base8internal30DstRangeRelationToSrcRangeImplIilLNS0_21IntegerRepresentationE1ELS2_1ELNS0_26NumericRangeRepresentationE0EE5CheckEl + 702092
18  libsystem_pthread.dylib             0x00007fff9045499d _pthread_body + 131
19  libsystem_pthread.dylib             0x00007fff9045491a _pthread_body + 0
20  libsystem_pthread.dylib             0x00007fff90452351 thread_start + 13
```

Any pointer to solving this is appreciated.

**Comment by deepak1556** [comment]
create the session once app is ready.

```
var {app, session} = require('electron')

app.on('ready', () => {
  let ses = session.defaultSession
})
```

**Comment by alchen** [comment]
Ah, yes. That's exactly what I was missing. Thank you.
